### PR TITLE
perf: improve SapQueryRunner performance

### PR DESCRIPTION
**Pull request by lotczyk**
Use async API exposed by driver

### Description of change

Usage of synchronous `prepare` was a critical bottleneck. In case of having appropriate parameters we can skip `prepare` execution altogether and call `exec` directly which can improve performance even further.

### Pull-Request Checklist

- [X] Code is up-to-date with the `master` branch
- [X] `npm run format` to apply prettier formatting
- [X] `npm run test` passes with this change
- [ ] This pull request links relevant issues as `Fixes #0000`
- [ ] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change N/A
- [X] The new commits follow conventions explained in [CONTRIBUTING.md](https://github.com/typeorm/typeorm/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: src/driver/sap/SapQueryRunner.ts [hunk 1]
```diff
@@ -202,15 +202,24 @@ export class SapQueryRunner extends BaseQueryRunner implements QueryRunner {
             const queryStartTime = +new Date()
             const isInsertQuery = query.substr(0, 11) === "INSERT INTO"
 
-            statement = databaseConnection.prepare(query)
+            if (parameters?.some(Array.isArray)) {
```
Comment (alumni) [inline src/driver/sap/SapQueryRunner.ts:205]: You might want to add a comment in the code why we have this condition.

File: src/driver/sap/SapQueryRunner.ts [hunk 2]
```diff
@@ -202,15 +202,24 @@ export class SapQueryRunner extends BaseQueryRunner implements QueryRunner {
...
-            const raw = await new Promise<any>((ok, fail) => {
-                statement.exec(parameters, (err: any, raw: any) =>
-                    err
-                        ? fail(new QueryFailedError(query, parameters, err))
-                        : ok(raw),
-                )
-            })
+            let raw: any
+            try {
+                raw = statement
```
Comment (alumni) [inline src/driver/sap/SapQueryRunner.ts:213]: This condition is identical to the single-branch if statement above, you could merge the two.

Comment (hinogi) [reply src/driver/sap/SapQueryRunner.ts:213]: there is a `prepare.bind` not an exec as I see it. But I do not know how they differ in execution. Therefore, if it is only a preparation for the exec and that only has to happen if there is some parameter in the parameters array this looks reasonable.

**Comment by alumni** [comment]
It would be nice to merge this, since I'd like to add streaming support and I'll also have to change the way the statements are prepared & executed.

**Comment by pleerock** [comment]
Thank you for contribution! 🎉

**Pull request merged**